### DISCUSSION TITLE
JSON layout does not display line numbers

### DISCUSSION THREAD

**Discussion by youngledo** [apache/logging-log4j2#2873] [Q&A]
## Description

Using JSON layout and configuring line numbers, but not in the file, but the console can display.

## Configuration

**Version:** 2.17.2

**Operating system:** macOS Sonoma 14.6

**JDK:** 8

## Logs

- File
  <img width="1056" alt="image" src="https://github.com/user-attachments/assets/554896e1-0257-49bf-8979-be8cdd9defb7">
- Console
  <img width="627" alt="image" src="https://github.com/user-attachments/assets/61b25080-b20a-499a-a091-1b6607fba88d">

  

## Reproduction
The example project is as follows:
[no-line-number.zip](https://github.com/user-attachments/files/16633678/no-line-number.zip)

**Comment by ppkarwasz** [comment]
JSON Template Layout does not print the location by default. You need to set its `locationInfoEnabled` attribute to `true` in order to enable it:
```xml
<JsonTemplateLayout eventTemplateUri="classpath:test.json"
                    locationInfoEnabled="true"/>
```

Due to #1692, you also need to upgrade to version 2.21.0 or later, since the bug prevents location information from being computed for some method calls.

**Comment by youngledo** [comment]
Also, can't JSON templates use the 'ThreadContext'?
<img width="365" alt="image" src="https://github.com/user-attachments/assets/be7855b6-2798-4581-ad20-cf448fe58880">

I found that the '%X' in the template is invalid.

**Comment by ppkarwasz** [comment]
> I found that the '%X' in the template is invalid.

See [this answer](https://github.com/apache/logging-log4j2/discussions/2872#discussioncomment-10407633).
